notebooks/text-classification_optuna.ipynb

```diff
@@ -0,0 +1,401 @@
+{
```

Comment:
<p>Since the original trainer uses <code>model_init</code> I think we can simplify this step to:</p><p> </p><pre class="ql-syntax" spellcheck="false">for k,v in best_run.hyperparameters.items():
    setattr(trainer, k, v) 
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/100/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
<p>That would not catch multi_target_strategy. Will have a look if I can make it a bit cleaner</p>

Reply:
@Mouhanedg56 @lewtun Refactored a bit to not have code duplication there. I think the interface is quite nice, however:

* It is diverging a little from transformers. In a good way I think, but could be surprising.
* Tests need a little refactoring if we go with this.

Reply:
Ah good point re `multi_target_strategy`!

I'm happy to diverge from the `transformers.Trainer` since it's only used as inspiration :) The current API in your PR looks neat to me, so let's use that + a small refactoring of the tests

Reply:
Great refactoring. The usage now is much easier than the previous version.